url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/compiler/XTreeCompiler.java

```diff
@@ -0,0 +1,1298 @@
+		}
+		visit(_try.body).asStatement(this, code);
+		code.addInstructionB(_try, XOpcode.POP, this.scope.getLocalsCount());
+		this.scope = this.scope.getParent();
+		code.addInstruction(_try, XOpcode.END_TRY);
+		if (_finally == null) {
+			code.addInstruction(_try, XOpcode.JUMP, _break);
+		} else {
+			code.addInstruction(_try, XOpcode.LOADN);
+			code.addInstruction(_try, XOpcode.LOADI_1);
+			code.addInstruction(_try, XOpcode.JUMP, _finally);
+		}
+		if (_try.catches != null) {
+			code.addInstruction(_try, onExc);
+			for (XTreeCatch _catch : _try.catches) {
+				XJumpTarget not = new XJumpTarget();
+				if (_catch.excType.size() == 1) {
+					code.addInstruction(_catch, XOpcode.DUP);
+					XCompiledPart part = visit(_catch.excType.get(0));
+					part.setupAndGet(this, code);
+					code.addInstruction(_catch, XOpcode.INSTANCEOF);
+					code.addInstruction(_catch, XOpcode.JUMP_IF_ZERO, not);
+				} else {
+					XJumpTarget is = new XJumpTarget();
+					for (XTreeExpr type : _catch.excType) {
+						code.addInstruction(_catch, XOpcode.DUP);
+						XCompiledPart part = visit(type);
+						part.setupAndGet(this, code);
+						code.addInstruction(_catch, XOpcode.INSTANCEOF);
+						code.addInstruction(_catch, XOpcode.JUMP_IF_NON_ZERO, is);
+					}
+					code.addInstruction(_catch, XOpcode.JUMP, not);
+					code.addInstruction(_catch, is);
+				}
+				this.scope = new XScope(this.scope, true);
+				if (_catch.varName == null) {
+					code.addInstruction(_catch, XOpcode.POP_1);
+				} else {
+					XCompiledPartVar part = new XCompiledPartVar(_catch, scope, _catch.varName, XVarAccess.LOCAL);
+					if (part.setup2(this, code)) {
+						throw new AssertionError();
+					}
+					part.set(this, code);
+				}
+				XCompiledPart part = visit(_catch.body);
+				part.asStatement(this, code);
+				code.addInstructionB(_catch, XOpcode.POP, this.scope.getLocalsCount());
+				this.scope = this.scope.getParent();
+				if (_finally == null) {
+					code.addInstruction(_try, XOpcode.JUMP, _break);
+				} else {
+					code.addInstruction(_try, XOpcode.LOADN);
+					code.addInstruction(_try, XOpcode.LOADI_1);
+					code.addInstruction(_try, XOpcode.JUMP, _finally);
+				}
+				code.addInstruction(_catch, not);
+			}
+			if (_finally == null) {
+				code.addInstruction(_try, XOpcode.POP_1);
+			}
+		}
+		if (_finally != null) {
+			code.addInstruction(_try, onFinallyExc);
+			code.addInstruction(_try, XOpcode.LOADI_0);
+			code.addInstruction(_try, _finally);
+			this.scope = new XScope(this.scope, true);
+			visit(_try._finally).asStatement(this, code);
+			code.addInstructionB(_try, XOpcode.POP, this.scope.getLocalsCount());
+			this.scope = this.scope.getParent();
+			code.addInstruction(_try, XOpcode.END_FINALLY);
+		}
+		this.scope = this.scope.getParent();
+		code.addInstruction(_try, _break);
+	}
+
+	@Override
+	public void visitGroup(XTreeGroup group) {
+		group.group.accept(this);
+	}
+
+	@Override
+	public void visitOperator(XTreeOperator operator) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 45. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.